mlflow/tensorflow/__init__.py

```diff
@@ -734,22 +799,21 @@ def predict(self, data):
 
 
 class _KerasModelWrapper:
-    def __init__(self, keras_model, graph, sess):
+    def __init__(self, keras_model, signature):
         self.keras_model = keras_model
-        self._graph = graph
-        self._sess = sess
+        self.signature = signature
 
     def predict(self, data):
-        def _predict(data):
-            if isinstance(data, pandas.DataFrame):
-                predicted = pandas.DataFrame(self.keras_model.predict(data.values))
-                predicted.index = data.index
-            else:
-                predicted = self.keras_model.predict(data)
-            return predicted
+        if self.signature is not None and isinstance(data, pandas.DataFrame):
+            # This line is for backwards compatibility.
```

Comment:
This comment sounds ambiguous. Can you add a few more details?

Reply:
Just found the line is wrong, it should be
`if self.signature is None and isinstance(data, pandas.DataFrame):`

Reply:
* If model signature is not None, then the `_enforce_schema` will convert dataframe input into dict input, so in the case the `_KerasModelWrapper.predict` will receive a dict type input.
* If model signature is None, the `_enforce_schema` can do nothing, and if the input is dataframe, the `_KerasModelWrapper.predict` will receive a dataframe input, we need handle this case, to keep backwards compatibility.